test/unit/search/helper.hpp

```diff
@@ -11,38 +11,31 @@
 #include <vector>
 
 #include <seqan3/alphabet/all.hpp>
+#include <seqan3/std/iterator>
 
 namespace seqan3
 {
 
-template <typename T>
-std::vector<T> uniquify(std::vector<T> v)
-{
-    std::sort(v.begin(), v.end());
-    v.erase(std::unique(v.begin(), v.end()), v.end());
-    return v;
-}
 
-template <typename T>
-std::vector<std::vector<T>> uniquify(std::vector<std::vector<T>> v)
+template <typename search_result_range_t>
+std::vector<std::ranges::range_value_t<search_result_range_t>> transform_to_vector(search_result_range_t && rng)
 {
-    std::for_each(v.begin(), v.end(), [] (auto & hits) { uniquify(hits); } );
-    return v;
-}
+    using hit_t = std::ranges::range_value_t<search_result_range_t>;
 
-template <typename T1, typename T2>
-std::vector<std::pair<T1, T2>> uniquify(std::vector<std::pair<T1, T2>> v)
-{
-    std::sort(v.begin(), v.end());
-    v.erase(std::unique(v.begin(), v.end()), v.end());
-    return v;
+    std::vector<hit_t> results{};
+    for (auto it = rng.begin(); it != rng.end(); ++it)
+        results.push_back(std::ranges::iter_move(it));
+
+    return results;
 }
 
-template <typename T1, typename T2>
-std::vector<std::vector<std::pair<T1, T2>>> uniquify(std::vector<std::vector<std::pair<T1, T2>>> v)
+template <typename search_result_range_t>
+std::vector<std::ranges::range_value_t<search_result_range_t>> uniquify(search_result_range_t && rng)
 {
-    std::for_each(v.begin(), v.end(), [] (auto & hits) { uniquify(hits); } );
-    return v;
+    auto unique_res = transform_to_vector(std::forward<search_result_range_t>(rng));
```

Comment:
The first part does not work, since the passed range is not a common range. For the reason of the second part, I don't know. That was before I left it because it is not part of this PR anyway. If you feel that is worth discussing you could put it on the strategy meeting.